internal/character/qingque/card.go

```diff
@@ -0,0 +1,144 @@
+package qingque
+
+func (c *char) swap(pos1, pos2 int) {
+	c.tiles[pos1], c.tiles[pos2] = c.tiles[pos2], c.tiles[pos1]
+	c.suits[pos1], c.suits[pos2] = c.suits[pos2], c.suits[pos1]
+}
+
+// Functions if distinct suits aren't needed to save time and memory, for documentation read the comments on the functions including distinct suits
+func (c *char) drawTile() string {
+	c.e2()
+	if c.tiles[0] != 4 {
+		s1, s2, s3 := c.tiles[0], c.tiles[1], c.tiles[2]
+		startingTiles := s1 + s2 + s3
+		drawn := c.engine.Rand().Intn(3)
+		c.tiles[drawn] += 1
+		if c.tiles[0] >= c.tiles[1] && c.tiles[1] >= c.tiles[2] {
+		} else if c.tiles[1] > c.tiles[0] {
+			c.swap(0, 1)
+		} else if c.tiles[2] > c.tiles[0] {
+			c.swap(0, 2)
+		} else {
+			c.swap(1, 2)
+		}
+		if startingTiles == 4 {
+			c.discardTile()
+		}
+	}
+	return "DISTINCT SUITS DISABLED"
+}
+func (c *char) discardTile() string {
+	if c.tiles[2] != 0 {
+		c.tiles[2] -= 1
+	} else if c.tiles[1] != 0 {
+		c.tiles[1] -= 1
+	} else if c.tiles[0] != 0 {
+		c.tiles[0] -= 1
+	} else {
+		// This should really never happen
+	}
```

Comment:
change the `if c.tiles[0] != 0` to be an else case so you don't need the "should never happen" else

Can also clean up a bit with a switch:
```go
switch {
case c.tiles[2] != 0:
  c.tiles[2] -= 1
case c.tiles[1] != 0:
  c.tiles[1] -= 1
default:
  c.tiles[0] -= 1
}
```